crates/bevy_pbr/src/atmosphere/mod.rs

```diff
@@ -0,0 +1,457 @@
+/// and the cosine of the zenith angle of the sun.
+///
+/// The sky-view lut is essentially the actual skybox, storing the light scattered
+/// towards the camera in every direction with a cubemap.
+///
+/// The aerial-view lut is a 3d LUT fit to the view frustum, which stores the luminance
+/// scattered towards the camera at each point (RGB channels), alongside the average
+/// transmittance to that point (A channel).
+#[derive(Clone, Component, Reflect, ShaderType)]
+pub struct AtmosphereSettings {
+    /// The size of the transmittance LUT
+    pub transmittance_lut_size: UVec2,
+
+    /// The size of the multiscattering LUT
+    pub multiscattering_lut_size: UVec2,
+
+    /// The size of the sky-view LUT.
+    pub sky_view_lut_size: UVec2,
+
+    /// The size of the aerial-view LUT.
+    pub aerial_view_lut_size: UVec3,
+
+    /// The number of points to sample along each ray when
+    /// computing the transmittance LUT
+    pub transmittance_lut_samples: u32,
+
+    /// The number of rays to sample when computing each
+    /// pixel of the multiscattering LUT
+    pub multiscattering_lut_dirs: u32,
+
+    /// The number of points to sample when integrating along each
+    /// multiscattering ray
+    pub multiscattering_lut_samples: u32,
+
+    /// The number of points to sample along each ray when
+    /// computing the sky-view LUT.
+    pub sky_view_lut_samples: u32,
+
+    /// The number of points to sample for each slice along the z-axis
+    /// of the aerial-view LUT.
+    pub aerial_view_lut_samples: u32,
+
+    /// The maximum distance from the camera to evaluate the
+    /// aerial view LUT. The slices along the z-axis of the
+    /// texture will be distributed linearly from the camera
+    /// to this value.
+    ///
+    /// units: m
+    pub aerial_view_lut_max_distance: f32,
+
+    /// A conversion factor between scene units and meters, used to
+    /// ensure correctness at different length scales.
+    pub scene_units_to_m: f32,
+}
+
+impl Default for AtmosphereSettings {
+    fn default() -> Self {
+        Self {
+            transmittance_lut_size: UVec2::new(256, 128),
+            transmittance_lut_samples: 40,
+            multiscattering_lut_size: UVec2::new(32, 32),
+            multiscattering_lut_dirs: 64,
+            multiscattering_lut_samples: 20,
+            sky_view_lut_size: UVec2::new(400, 200),
+            sky_view_lut_samples: 16,
+            aerial_view_lut_size: UVec3::new(32, 32, 32),
+            aerial_view_lut_samples: 10,
+            aerial_view_lut_max_distance: 3.2e4,
+            scene_units_to_m: 1.0,
+        }
+    }
+}
+
+impl ExtractComponent for AtmosphereSettings {
+    type QueryData = Read<AtmosphereSettings>;
+
+    type QueryFilter = (With<Camera3d>, With<Atmosphere>);
+
+    type Out = AtmosphereSettings;
+
+    fn extract_component(item: QueryItem<'_, Self::QueryData>) -> Option<Self::Out> {
+        Some(item.clone())
+    }
+}
+
+fn configure_camera_depth_usages(
```

Comment:
I really dislike how we have that system in a bunch of different places. Don't need to do anything in this PR but I think there's something a bit weird with the fact that we need it in so many places.